athena-jdbc/src/main/java/com/amazonaws/connectors/athena/jdbc/manager/JdbcMetadataHandler.java

```diff
@@ -138,12 +138,20 @@ public ListSchemasResponse doListSchemaNames(final BlockAllocator blockAllocator
     private Set<String> listDatabaseNames(final Connection jdbcConnection)
             throws SQLException
     {
-        try (ResultSet resultSet = jdbcConnection.getMetaData().getSchemas()) {
+        DatabaseMetaData metadata = jdbcConnection.getMetaData();
+        String databaseproductname = metadata.getDatabaseProductName();
+        try (ResultSet resultSet = metadata.getSchemas()) {
             ImmutableSet.Builder<String> schemaNames = ImmutableSet.builder();
             while (resultSet.next()) {
                 String schemaName = resultSet.getString("TABLE_SCHEM");
-                // skip internal schemas
-                if (!schemaName.equals("information_schema")) {
+                
+                //skip internal schemas for SAP HANA Database
+                if (databaseproductname != null && !databaseproductname.isEmpty() && databaseproductname.equals("HDB")) {
+                    if (!schemaName.startsWith("_") && !schemaName.contains("SYS") && !schemaName.contains("XS") && !schemaName.equals("SAPDBCTRL") && !schemaName.contains("SAP_") && !schemaName.contains("UIS")) {
+                        schemaNames.add(schemaName);
+                    }
```

Comment:
This seems like too much of a special case. Why do we need to do this? and why can't it be handled in the SAP specific code?

Reply:
Yes, this can be handled at application level. Updated.